tests/unit/core/test_utils.js

```diff
@@ -209,6 +209,48 @@ describe('getClientApp', () => {
   });
 });
 
+describe('isAddonAuthor', () => {
+  const addon = {
+    ...fakeAddon,
+    authors: [
+      {
+        id: 5838591,
+        name: 'tofumatt',
+        picture_url: 'http://cdn.a.m.o/myphoto.jpg',
+        url: 'http://a.m.o/en-GB/firefox/user/tofumatt/',
+        username: 'tofumatt',
+      },
+    ],
+  };
+
+  it('returns true when userId is in add-on author object', () => {
+    expect(isAddonAuthor({ addon, userId: 5838591 })).toEqual(true);
+  });
+
+  it('returns true when userId is not in add-on author object', () => {
+    expect(isAddonAuthor({ addon, userId: 5838592 })).toEqual(false);
+  });
+
+  it('returns false when addon is not set', () => {
+    expect(isAddonAuthor({ addon: null, userId: 5838591 })).toEqual(false);
```

Comment:
Since the code relies on falsyness (not just null) then I think you need another test for the `addon: undefined` case. 

Ideally the function should throw if `addon` or `userId` is undefined (since it could have been a calling mistake) but I think React might actually set `addon` to undefined in some cases because JavaScript is awful.